src/Interpreters/executeQuery.cpp

```diff
@@ -865,6 +867,44 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
                     if (auto query_log = context->getQueryLog())
                         query_log->add(elem);
                 }
+                if (log_processors_profiles)
+                {
+                    if (auto processors_profile_log = context->getProcessorsProfileLog())
+                    {
+                        ProcessorProfileLogElement processor_elem;
+                        processor_elem.event_time = time_in_seconds(finish_time);
+                        processor_elem.event_time_microseconds = time_in_microseconds(finish_time);
+                        processor_elem.query_id = elem.client_info.current_query_id;
+
+                        auto get_proc_id = [](const IProcessor & proc) -> UInt64
+                        {
+                            return reinterpret_cast<std::uintptr_t>(&proc);
+                        };
```

Comment:
Maybe it would be good to re-enumerate processors, so that ids were in [0 ... N - 1]
I thought about it implementing `explain pipeline`, but newer did.

Maybe we can do it in both places, so that ids from explain will likely be the same

Reply:
Will do this separately to avoid bloating this patch set.